app/controllers/application_controller.rb

```diff
@@ -39,20 +40,19 @@ def check_maintenance_mode_is_enabled
     end
   end
 
-  def append_info_to_payload(payload)
-    super
-    payload[:host] = request.host
+  def set_logging_attributes
+    CurrentLoggingAttributes.host = request.host
+    CurrentLoggingAttributes.request_id = request.request_id
+    CurrentLoggingAttributes.session_id_hash = Digest::SHA256.hexdigest session.id.to_s if session.exists?
```

Comment:
If `session.exists?` then is it certain that `session.id` exists? Otherwise we'll get a NPE on `session.id.to_s`, Might be safer to use the safe reference like `session.id&.to_s`. I can see this is copied from the existing but worth checking I think. My concern with common logging is that it's invoked on every request and if its wrong then it can cause unfortunate issues.

Reply:
Yeh that's a good point, I'll make that change to make it safer